Library/Homebrew/test/utils/git_spec.rb

```diff
@@ -0,0 +1,175 @@
+                                                    before_commit: hash2),
+      ).to eq(hash2)
+    end
+  end
+
+  describe "#last_revision_of_file" do
+    it "returns last revision of file" do
+      expect(
+        described_class.last_revision_of_file(@repo,
+                                              @repo/@file),
+      ).to eq("blah")
+    end
+
+    it "returns last revision of file based on before_commit" do
+      expect(
+        described_class.last_revision_of_file(@repo, @repo/@file,
+                                              before_commit: "0..3"),
+      ).to eq("brew")
+    end
+  end
+end
+
+describe Utils do
+  before(:each) do
+    if described_class.instance_variable_defined?(:@git)
+      described_class.send(:remove_instance_variable, :@git)
+    end
+  end
+
+  describe "::git_available?" do
+    it "returns true if git --version command succeeds" do
+      expect(described_class.git_available?).to be_truthy
+    end
+
+    it "returns false if git --version command does not succeed" do
+      stub_const("HOMEBREW_SHIMS_PATH", mktmpdir/"shim")
+      expect(described_class.git_available?).to be_falsey
+    end
+  end
+
+  describe "::git_path" do
+    it "returns nil when git is not available" do
+      stub_const("HOMEBREW_SHIMS_PATH", mktmpdir/"shim")
+      expect(described_class.git_path).to eq(nil)
+    end
+
+    it "returns path of git when git is available" do
+      expect(described_class.git_path).to end_with("git")
+    end
+  end
+
+  describe "::git_version" do
+    it "returns nil when git is not available" do
+      stub_const("HOMEBREW_SHIMS_PATH", mktmpdir/"shim")
+      expect(described_class.git_path).to eq(nil)
+    end
+
+    it "returns version of git when git is available" do
+      expect(described_class.git_version).not_to be_nil
+    end
+  end
+
+  describe "::ensure_git_installed!" do
+    it "returns nil if git already available" do
+      expect(described_class.ensure_git_installed!).to be_nil
+    end
+
+    context "when git is not already available" do
+      before do
+        stub_const("HOMEBREW_SHIMS_PATH", mktmpdir/"shim")
+      end
+
+      it "can't install brewed git if homebrew/core is unavailable" do
+        allow_any_instance_of(Pathname).to receive(:directory?).and_return(false)
+        expect { described_class.ensure_git_installed! }.to raise_error("Git is unavailable")
+      end
+
+      it "raises error if can't install git" do
+        stub_const("HOMEBREW_BREW_FILE", mktmpdir/"brew")
+        expect { described_class.ensure_git_installed! }.to raise_error("Git is unavailable")
+      end
+
+      it "installs git" do
+        allow(Homebrew).to receive(:_system).with(any_args).and_return(true)
+        described_class.ensure_git_installed!
+      end
+    end
+  end
+
+  describe "::git_remote_exists" do
+    it "returns true when git is not available" do
+      stub_const("HOMEBREW_SHIMS_PATH", mktmpdir/"shim")
```

Comment:
Instead of `mktmpdir` write to an already existing location that's deleted after tests: for example: https://github.com/Homebrew/brew/blob/2c43e069c25e3c7cc255f5cd105df3597246b952/Library/Homebrew/test/spec_helper.rb#L110-L129